### PR TITLE
SQLiteTypeMapping<T>, ContentResolverTypeMapping<T> generation

### PR DESCRIPTION
**Pull request by zayass**
Hi

I think this is to hard way for operate with generated code
```java
.addTypeMapping(Tweet.class, SQLiteTypeMapping.<Tweet>builder()
        .putResolver(new TweetStorIOSQLitePutResolver())
        .getResolver(new TweetStorIOSQLiteGetResolver())
        .deleteResolver(new TweetStorIOSQLiteDeleteResolver())
        .build())
.addTypeMapping(User.class, SQLiteTypeMapping.<User>builder()
        .putResolver(new UserStorIOSQLitePutResolver())
        .getResolver(new UserStorIOSQLiteGetResolver())
        .deleteResolver(new UserStorIOSQLiteDeleteResolver())
        .build())
.addTypeMapping(TweetWithUser.class, SQLiteTypeMapping.<TweetWithUser>builder()
        .putResolver(new TweetWithUserPutResolver())
        .getResolver(new TweetWithUserGetResolver())
        .deleteResolver(new TweetWithUserDeleteResolver())
        .build())
```

Provided code generate extra classes for collapse it to
```java
.addTypeMapping(Tweet.class, new TweetSQLiteTypeMapping())
.addTypeMapping(User.class, new UserSQLiteTypeMapping())
```

### CODE REVIEW

**Comment by artem-zinnatullin** [comment]
Nice idea!

@nikitin-da can you please take a look?

File: storio-content-resolver-annotations-processor/src/main/java/com/pushtorefresh/storio/contentresolver/annotations/processor/generate/MappingGenerator.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+                .addMethod(createConstructor(storIOSQLiteTypeMeta))
+                .build();
+
+        return JavaFile
+                .builder(storIOSQLiteTypeMeta.packageName, mapping)
+                .indent(INDENT)
+                .build();
+    }
+
+    private MethodSpec createConstructor(StorIOContentResolverTypeMeta storIOSQLiteTypeMeta) {
```
Comment (artem-zinnatullin) [inline storio-content-resolver-annotations-processor/src/main/java/com/pushtorefresh/storio/contentresolver/annotations/processor/generate/MappingGenerator.java]: Please add nullability annotations

File: storio-sample-app/src/main/java/com/pushtorefresh/storio/sample/db/DbModule.java [hunk 2]
```diff
@@ -39,16 +35,8 @@
...
-                        .putResolver(new TweetStorIOSQLitePutResolver())
-                        .getResolver(new TweetStorIOSQLiteGetResolver())
-                        .deleteResolver(new TweetStorIOSQLiteDeleteResolver())
-                        .build())
-                .addTypeMapping(User.class, SQLiteTypeMapping.<User>builder()
-                        .putResolver(new UserStorIOSQLitePutResolver())
-                        .getResolver(new UserStorIOSQLiteGetResolver())
-                        .deleteResolver(new UserStorIOSQLiteDeleteResolver())
-                        .build())
+                .addTypeMapping(Tweet.class, new TweetSQLiteTypeMapping())
```
Comment (artem-zinnatullin) [inline storio-sample-app/src/main/java/com/pushtorefresh/storio/sample/db/DbModule.java]: :+1:

File: storio-sample-app/src/main/java/com/pushtorefresh/storio/sample/db/DbModule.java [hunk 3]
```diff
@@ -39,16 +35,8 @@
...
+                .addTypeMapping(User.class, new UserSQLiteTypeMapping())
```
Comment (artem-zinnatullin) [inline storio-sample-app/src/main/java/com/pushtorefresh/storio/sample/db/DbModule.java]: Can you please leave it as is to show that user may also specify resolvers independently? Probably worth adding a comment about it.

File: storio-sample-app/src/main/java/com/pushtorefresh/storio/sample/db/entities/Tweet.java [hunk 4]
```diff
@@ -14,6 +17,7 @@
 // Which will generate type mapping code in compile time,
 // You just need to link it in your code.
 @StorIOSQLiteType(table = TweetsTable.TABLE)
+@StorIOContentResolverType(uri = TweetMeta.URI_STRING)
```
Comment (artem-zinnatullin) [inline storio-sample-app/src/main/java/com/pushtorefresh/storio/sample/db/entities/Tweet.java]: I'm afraid a lot of people will be confused by both annotations on same class, but I think I'm okay to merge it and then I'll just add separate sample app for contentresolver

File: storio-sqlite-annotations-processor/src/main/java/com/pushtorefresh/storio/sqlite/annotations/processor/generate/MappingGenerator.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+                .addMethod(createConstructor(storIOSQLiteTypeMeta))
+                .build();
+
+        return JavaFile
+                .builder(storIOSQLiteTypeMeta.packageName, mapping)
+                .indent(INDENT)
+                .build();
+    }
+
+    private MethodSpec createConstructor(StorIOSQLiteTypeMeta storIOSQLiteTypeMeta) {
```
Comment (artem-zinnatullin) [inline storio-sqlite-annotations-processor/src/main/java/com/pushtorefresh/storio/sqlite/annotations/processor/generate/MappingGenerator.java]: Nullability annotations

File: storio-sqlite-annotations-processor/src/main/java/com/pushtorefresh/storio/sqlite/annotations/processor/generate/PutResolverGenerator.java [hunk 6]
```diff
@@ -19,13 +19,19 @@
 import static javax.lang.model.element.Modifier.PROTECTED;
 import static javax.lang.model.element.Modifier.PUBLIC;
 
-public class PutResolverGenerator implements ResolverGenerator<StorIOSQLiteTypeMeta> {
+public class PutResolverGenerator implements Generator<StorIOSQLiteTypeMeta> {
+
+    public static final String SUFFIX = "StorIOSQLitePutResolver";
+
+    public static String generateName(@NotNull StorIOSQLiteTypeMeta storIOSQLiteTypeMeta) {
```
Comment (artem-zinnatullin) [inline storio-sqlite-annotations-processor/src/main/java/com/pushtorefresh/storio/sqlite/annotations/processor/generate/PutResolverGenerator.java]: nullability

**Comment by artem-zinnatullin** [comment]
Wait wait wait, please don't merge master to this branch, use `rebase`, Luke! (Or I'll rebase it and merge manually :)

File: storio-sample-app/src/main/java/com/pushtorefresh/storio/sample/db/entities/Tweet.java (hunk 4 above)
Comment (zayass) [inline storio-sample-app/src/main/java/com/pushtorefresh/storio/sample/db/entities/Tweet.java]: Maybe. But I see Resolvers like com.pushtorefresh.storio.sample.provider.meta.TweetMeta#GET_RESOLVER and confused too. Why its not generated?

Comment (artem-zinnatullin) [inline storio-sample-app/src/main/java/com/pushtorefresh/storio/sample/db/entities/Tweet.java]: Yep, valid point :sweat_smile: We need to create clean separate examples of working with StorIO

**Comment by artem-zinnatullin** [comment]
:+1: 

What about rebase and squash into one commit? Or you can leave it as is and I'll do it myself when we'll got :+1: from @nikitin-da

**Comment by nikitin-da** [comment]
Cool feature! LGTM

**Comment by nikitin-da** [comment]
@zayass Sorry, but can you please rebase on latest master again?

**Comment by zayass** [comment]
@nikitin-da rebased

**Comment by nikitin-da** [comment]
Thanks

**Pull request merged**